src/include/OpenImageIO/errorhandler.h

```diff
@@ -83,16 +83,37 @@ class OIIO_API ErrorHandler {
     }
     virtual ~ErrorHandler() {}
 
+    /// Set desired verbosity level.
+    void verbosity(int v) { m_verbosity = v; }
+
+    /// Return the current verbosity level.
+    int verbosity() const { return m_verbosity; }
+
     /// The main (or "full detail") method -- takes a code (with high
     /// bits being an ErrCode) and writes the message, with a prefix
     /// indicating the error category (no prefix for "MESSAGE") and
     /// error string.
     virtual void operator()(int errcode, const std::string& msg);
 
-    /// Info message with printf-like formatted error message.
-    /// Will not print unless verbosity >= VERBOSE.
+    // Base cases -- take a single string
+    void info(const std::string& msg) { (*this)(EH_INFO, msg); }
+    void warning(const std::string& msg) { (*this)(EH_WARNING, msg); }
+    void error(const std::string& msg) { (*this)(EH_ERROR, msg); }
+    void severe(const std::string& msg) { (*this)(EH_SEVERE, msg); }
+    void message(const std::string& msg) { (*this)(EH_MESSAGE, msg); }
+#ifndef NDEBUG
+    void debug(const std::string& msg) { (*this)(EH_DEBUG, msg); }
+#else
+    void debug(const std::string&) {}
+#endif
+
+    //
+    // Formatted output with the same notation as Strutil::format.
+    /// Use with caution! Some day this will change to be fmt-like rather
+    /// than printf-like.
+    //
     template<typename... Args>
-    void info(string_view format, const Args&... args)
+    void info(const char* format, const Args&... args)
     {
         if (verbosity() >= VERBOSE)
             info(Strutil::format(format, args...));
```

Comment:
Would there be a way to avoid the allocation here?

Reply:
Which allocation? The string produced by the format() call?